libs/langchain-core/src/utils/json_schema.ts

```diff
@@ -16,22 +16,30 @@ import {
 
 export { deepCompareStrict, Validator } from "@cfworker/json-schema";
 
+export type ToJSONSchemaParams = NonNullable<
+  Parameters<typeof toJSONSchema>[1]
+>;
+
 /**
  * Converts a Zod schema or JSON schema to a JSON schema.
  * @param schema - The schema to convert.
+ * @param _params - The parameters to pass to the toJSONSchema function.
  * @returns The converted schema.
  */
-export function toJsonSchema(schema: InteropZodType | JSONSchema): JSONSchema {
+export function toJsonSchema(
+  schema: InteropZodType | JSONSchema,
+  _params?: ToJSONSchemaParams
```

Comment:
Since we are using `params` in the function, no need to have it start with `_`
```suggestion
  params?: ToJSONSchemaParams
```

Reply:
Fair point. I only named it such as that's what zod's method def uses:

```
export declare function toJSONSchema(schema: schemas.$ZodType, _params?: ToJSONSchemaParams):
```

But I agree we should just use without the `_` here.